api/api/src/main/scala/uk/ac/wellcome/platform/api/QueryParams.scala

```diff
@@ -0,0 +1,227 @@
+package uk.ac.wellcome.platform.api
+
+import java.time.LocalDate
+import io.circe.Decoder
+import io.circe.java8.time.TimeInstances
+import akka.http.scaladsl.server.{Directives, ValidationRejection}
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import io.circe.{Decoder, Json}
+
+import uk.ac.wellcome.platform.api.services.WorksSearchOptions
+import uk.ac.wellcome.platform.api.models._
+import uk.ac.wellcome.display.models._
+
+sealed trait QueryParams
+
+case class SingleWorkParams(
+  include: Option[V2WorksIncludes],
+  _index: Option[String],
+) extends QueryParams
+
+object SingleWorkParams extends QueryParamsUtils {
+
+  def parse =
+    parameter(
+      (
+        "include".as[V2WorksIncludes].?,
+        "_index".as[String].?
+      )
+    ).tmap((SingleWorkParams.apply _).tupled(_))
+}
+
+case class MultipleWorksParams(
+  page: Option[Int],
+  pageSize: Option[Int],
+  workType: Option[WorkTypeFilter],
+  `items.locations.locationType`: Option[ItemLocationTypeFilter],
+  `production.dates.from`: Option[LocalDate],
+  `production.dates.to`: Option[LocalDate],
+  language: Option[LanguageFilter],
+  `genres.label`: Option[GenreFilter],
+  `subjects.label`: Option[SubjectFilter],
+  include: Option[V2WorksIncludes],
+  aggregations: Option[List[AggregationRequest]],
+  sort: Option[List[SortRequest]],
+  sortOrder: Option[SortingOrder],
+  query: Option[String],
+  _queryType: Option[String],
+  _index: Option[String],
+) extends QueryParams {
+
+  def searchOptions(apiConfig: ApiConfig): WorksSearchOptions =
+    WorksSearchOptions(
+      filters = filters,
+      pageSize = pageSize.getOrElse(apiConfig.defaultPageSize),
+      pageNumber = page.getOrElse(1),
+      aggregations = aggregations.getOrElse(Nil),
+      sortBy = sort.getOrElse(Nil),
+      sortOrder = sortOrder.getOrElse(SortingOrder.Ascending),
+    )
+
+  def workQuery: Option[WorkQuery] =
+    query.map { qry =>
+      _queryType.map(_.toLowerCase) match {
+        case _ => WorkQuery.MSMBoostQuery(qry)
+      }
+    }
+
+  def validationErrors: List[String] =
+    List(
+      page
+        .filterNot(_ >= 1)
+        .map(_ => "page: must be greater than 1"),
```

Comment:
Annoyingly the error here is between 1 - 10000.
https://api.wellcomecollection.org/catalogue/v2/works?page=10000

Reply:
This is actually caught by the error handler [here](https://github.com/wellcometrust/catalogue/blob/master/api/api/src/main/scala/uk/ac/wellcome/platform/api/elasticsearch/ElasticErrorHandler.scala#L38-L40), with a 400 being returned. This is how we deal with this currently, can potentially catch it earlier in a subsequent PR but for now lets just leave the current behaviour.